ibis/expr/types/generic.py

```diff
@@ -2223,9 +2223,23 @@ class NullColumn(Column, NullValue):
 
 
 @public
-def null():
-    """Create a NULL/NA scalar."""
-    return ops.NULL.to_expr()
+def null(type: dt.DataType | str | None = None) -> Value:
+    """Create a NULL/NA scalar.
+
+    By default, the type will be NULLTYPE. This is castable and comparable to any type,
+    but lacks datatype-specific methods:
+
+    >>> import ibis
+    >>> ibis.null().upper().execute() is None
+    Traceback (most recent call last):
+        ...
+    AttributeError: 'NullScalar' object has no attribute 'upper'
+    >>> ibis.null(str).upper().execute() is None
+    True
```

Comment:
```suggestion
    >>> ibis.null(str).upper()
    None
```

Reply:
This needs to stay the same.

Without the .execute(), this just shows the compiled repr: `Uppercase(None): Uppercase(None)`

and if you repr the None directly, it doesn't show up in the output:

```
>>> ibis.null(str).upper().execute()
<nothing here>
```

Therefore I added the `is None` check so you actually see something.

Reply:
Since you're calling `execute` anyway, just set `ibis.options.interactive = True` first and then the value will repr as `None`